src/django_components/slots.py

```diff
@@ -227,75 +395,181 @@ def __init__(
         kwargs: RuntimeKwargs,
         trace_id: str,
         node_id: Optional[str] = None,
-        is_implicit: bool = False,
     ):
         super().__init__(nodelist=nodelist, args=None, kwargs=kwargs, node_id=node_id)
 
-        self.is_implicit = is_implicit
         self.trace_id = trace_id
-        self.component_id: Optional[str] = None
 
     def render(self, context: Context) -> str:
+        if self._is_extracting_fill(context):
+            self._extract_fill(context)
+            return ""
+
         raise TemplateSyntaxError(
-            "{% fill ... %} block cannot be rendered directly. "
-            "You are probably seeing this because you have used one outside "
-            "a {% component %} context."
+            "FillNode.render() (AKA {% fill ... %} block) cannot be rendered outside of a Component context. "
+            "Make sure that the {% fill %} tags are nested within {% component %} tags."
         )
 
     def __repr__(self) -> str:
-        return f"<{type(self)} ID: {self.node_id}. Contents: {repr(self.nodelist)}.>"
+        return f"<{self.__class__.__name__} ID: {self.node_id}. Contents: {repr(self.nodelist)}.>"
 
-    def resolve_kwargs(self, context: Context, component_name: Optional[str] = None) -> Dict[str, Optional[str]]:
+    def resolve_kwargs(self, context: Context) -> "FillWithData":
         kwargs = self.kwargs.resolve(context)
 
-        name = self._resolve_kwarg(kwargs, SLOT_NAME_KWARG, "slot name", component_name, identifier=False)
-        default_key = self._resolve_kwarg(kwargs, SLOT_DEFAULT_KWARG, "slot default", component_name)
-        data_key = self._resolve_kwarg(kwargs, SLOT_DATA_KWARG, "slot data", component_name)
+        name = self._process_kwarg(kwargs, SLOT_NAME_KWARG, identifier=False)
+        default_var = self._process_kwarg(kwargs, SLOT_DEFAULT_KWARG)
+        data_var = self._process_kwarg(kwargs, SLOT_DATA_KWARG)
+
+        # Allow to target the default slot by not providing a name, but still using `{% fill %}`
+        # tag, so one can access the `data` and `default` vars.
+        # E.g.
+        # ```django
+        # {% component "mytable" %}
+        #   {% fill default="default" data="data" %}
+        #     {{ data.some_var }}
+        #   {% endfill %}
+        # {% endcomponent %}
+        # ```
+        if name is None:
```

Comment:
NOTE: If we decided to roll back the feature of allowing to specify the default fill by omitting the `name` kwarg, then we'd also remove this if branch